.github/workflows/ci.yml

```diff
@@ -1,19 +1,38 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+    branches:
+      - main
 
 env:
   CARGO_TERM_COLOR: always
 
 jobs:
+  cancel_running_workflows:
+    name: Cancel running workflows
+    runs-on: ubuntu-20.04
+    steps:
+      - name: cancel running workflows
+        uses: styfle/cancel-workflow-action@0.8.0
+        with:
+          access_token: ${{ github.token }}
```

Comment:
Can someone explain to me how running cancellation works? Is it per PR or global?

Reply:
Thanks for explaining. Per PR/branch is good.